feat(Forms): auto-open iterate container items when validation errors and make `Iterate.Toolbar` fully customizable (#3877)
dnbexperience/eufemia: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Toolbar/info.mdx

)
```

> ## Cusomize the Toolbar

Reviewer: ```suggestion
## Customize the Toolbar
```

)
```

> ## Cusomize the Toolbar

Reviewer: ```suggestion
## Customize the Toolbar
```

You can customize the toolbar by passing a function as a child to `Iterate.Toolbar`. The function receives the following parameters as an object:

- `index` the index of the current item in the array.
- `items` the array of items.
- `EditButton` used in the `Iterate.ViewContainer`.
- `RemoveButton` used in the `Iterate.ViewContainer`.
- `CancelButton` used in the `Iterate.EditContainer`.
- `DoneButton` used in the `Iterate.EditContainer`.

```tsx
import { Iterate } from '@dnb/eufemia/extensions/forms'

render(
  <Iterate.Array>
    <Iterate.ViewContainer>
      Item Content
      <Iterate.Toolbar>
        {({
          index,
          items,
          EditButton,
          RemoveButton,
          CancelButton,
          DoneButton,
        }) => {
          if (items.length === 1) {
            return null
          }

          return (
            <>
              <EditButton />
              <RemoveButton />
            </>
          )
        }}
      </Iterate.Toolbar>
    </Iterate.ViewContainer>
  </Iterate.Array>,
)
```

## Variants

### `minimumRequiredItems`

This variant has the following behavior:

> - When `EditContainer` is visible, and the number of items in the array is one, the entier toolbar will be hidden.

Reviewer: ```suggestion
- When `EditContainer` is visible, and the number of items in the array is one, the entire toolbar will be hidden.
```

> - When `ViewContainer` is visible, and the number of items in the array is one, the delete/remove button will be hidden.

Reviewer: ```suggestion
- When `ViewContainer` is visible, and the number of items in the array is one, theremove button will be hidden.
```

I think it's safe to use remove button, as that's the name of the component as well `<Iterate.RemoveButton />`
